contracts/api3-server-v1/proxies/CompositeApi3ReaderProxyV1.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import "./interfaces/ICompositeApi3ReaderProxyV1.sol";
+import "../../interfaces/IApi3ReaderProxy.sol";
+
+import "hardhat/console.sol";
+
+/// @title An immutable proxy contract that is used to read a composite data feed
+/// by multiplying two IApi3ReaderProxy data feeds
+/// @dev This contract implements the AggregatorV2V3Interface to be compatible with
+/// Chainlink aggregators. This allows the contract to be used as a drop-in
+/// replacement for Chainlink aggregators in existing dApps.
+/// Refer to https://github.com/api3dao/migrate-from-chainlink-to-api3 for more
+/// information about the Chainlink interface implementation.
+contract CompositeApi3ReaderProxyV1 is ICompositeApi3ReaderProxyV1 {
+    /// @notice First IApi3ReaderProxy contract address
+    address public immutable override proxy1;
+
+    /// @notice Second IApi3ReaderProxy contract address
+    address public immutable override proxy2;
+
+    /// @notice Decimals of the rate composition
+    uint8 public immutable override decimals;
+
+    uint256 private immutable baseUnit;
```

Comment:
This doesn't work. In the implementation you note `IApi3ReaderProxy always return values in 18 decimals` but here you allow the proxies to have different decimals.

I think there should be a separate contract for scaling the decimals precision. That way you can combine the proxies and at the end wrap it in `ScaledApi3FeedProxyV1` (I'd intentionally avoid choose a different naming to make it more apparent this is not compatible with proxies). I guess this one could depend on `AggregatorV2V3Interface` and be used to convert the non-Api3 proxies to `AggregatorV2V3Interface`.

Reply:
This is a valid point, I should have said that `Api3ReaderProxyV1` proxies always return 18 decimals but it still breaks when either `proxy1` or `proxy2` don't return 18 decimals. I guess we can't even expect `proxy1` and `proxy2` to be `AggregatorV2V3Interface` because they might even be old legacy `IProxy` proxies 🤔 I guess having a wrapper scaling contract to be used at the end sounds like a good idea 👍🏻

Reply:
Another alternative could be to set the decimals for each proxy via constructor arguments and keep the `baseUnit` scaling

Reply:
https://github.com/api3dao/contracts/pull/472/commits/0cb5750059430cf8fc86786f89a58dff8920b6a8